iocore/aio/I_AIO.h

```diff
@@ -70,10 +70,10 @@ typedef struct io_event ink_io_event_t;
 #else
 
 struct ink_aiocb {
-  int aio_fildes         = 0;
-  volatile void *aio_buf = nullptr; /* buffer location */
```

Comment:
Shouldn't this be an atomic?

Reply:
All this patch is doing is removing volatile, which did not give us thread safety. So this shouldn't really change anything with regard to that. If it *should* be atomic, that might be another issue.

Reply:
@PSUdaemon We should investigate why each of the variables were volatile and covert them to atomic in this PR.  Having the changes to atomics in another PR would make it harder to see what needs/did to change to atomic.

Reply:
@bryancall Should I do the conversions with `std::atomic` or `ink_atomic`? I think I'd prefer the former, especially if we plan to remove the latter soon.

Reply:
@SolidWallOfCode I don't think `aio_buf` needs to be an atomic. It's protected by a mutex in all usage.

Reply:
@SolidWallOfCode I don't think `aio_buf` needs to be an atomic. Can you explain why you think otherwise?

Reply:
@bryancall a lot of them are really wrong usage. I fixed a signal handler that was not using atomics as it should have been, and an example that had a comment that said "use a mutex". The rest seems like FUD usage. Other than that, everything looked good in that if it weren't it's so totally wrong it could never have worked in the first place AFAICT.

Reply:
@PSUdaemon I would use std::atomic

Reply:
Good, cause that's what I did :)

Reply:
I think he means on that line. Is this variable shared between threads?

Reply:
@dragon512 no, it's just a member of a struct for AIO. Each AIO operation should have its own struct. There is a lot of "Oh, we are going to update this then something else has to read it. Better make it volatile so the other thing sees the new data" in the code. Totally useless AFAICT.

Reply:
Ya this should be removed... it slowing the code down by forcing register load in the pipeline.